site/_posts/2020-07-20-breached-password-detection.md

```diff
@@ -0,0 +1,148 @@
+
+If you are going to build this feature, look for available datasets. You can also [look for APIs](https://haveibeenpwned.com/API/v3) to help you. Consider how to ingest and expose these sets to your auth systems in a performant manner. You'll also need to make sure you schedule time to keep these datasets up to date. 
+
+When building this feature, build a way to force users to change their compromised passwords. There's not a ton of literature out there, but [this recent study (PDF)](https://www.ieee-security.org/TC/SPW2020/ConPro/papers/bhagavatula-conpro20.pdf) of almost 250 users over two years showed that only 13% of users with a breached password changed their credentials within three months. From the conclusion:
+
+> Overall, our findings suggest that password breach notifications are failing dramatically, both at causing users to take action and at causing users to take *constructive* action. Regulators should take note of the ineffectiveness or absence of breach notifications and impose requirements on companies to implement better practices.
+
+Ouch.
+
+## How FusionAuth detects breached passwords
+
+*Please note that this is a paid edition feature. You can [learn more about paid editions here](/pricing).*
+
+We have built breached password detection into our auth management system. Enabling real time compromised password detection and mitigation in FusionAuth couldn't be easier. 
+
+Assuming you've [activated your paid edition with a license key](/docs/v1/tech/reactor), navigate to the Tenant details page, and then to the "Password" tab. Enable "Breached password detection settings" and then choose your options. If you need varied configuration for different applications, use the multi-tenant feature to create separate tenants and configure them individually.
+
+{% include _image.liquid src="/assets/img/blogs/breached-password-detection/enable-breached-password-check.png" alt="Enabling breached password detection." class="img-fluid" figure=false %}
+
+This tab is also where you can specify various password settings. The defaults are fine, but feel free to change the rules to meet your needs, whether requiring certain types of characters, setting an expiration time, or mandating a minimum length. Just don't make me hop on one foot, please.
+
+When configuring breached password detection, there are three levels of compromised password matching. You can match:
+
+* on the password alone
+* on the password plus username, email address or email sub-address
+* on the password and an exact match on username or email address, or commonly used passwords like "password". 
+
+While you know your security requirements best, we recommend matching on password alone, as this provides the most protection. 
+
+Matching rules apply to all passwords provided whenever a user is created. This includes:
```

Comment:
I'm not clear on this section. The matching rules are applied anytime the user touches their password correct? Doesn't this include creation, update, and password change?

Reply:
Thanks, made it clearer. I just tested and when you update a user (but don't change their password) the check is not performed. But anytime a password is touched, the check is performed.